pkg/cloud/ibmcloud/clients/machines.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2018 The Kubernetes authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clients
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/softlayer/softlayer-go/datatypes"
+	"github.com/softlayer/softlayer-go/services"
+	"github.com/softlayer/softlayer-go/session"
+	"github.com/softlayer/softlayer-go/sl"
+)
+
+type GuestService struct {
+	sess *session.Session
+}
+
+func NewGuestService(sess *session.Session) GuestService {
+	return GuestService{sess: sess}
+}
+
+func (gs *GuestService) guestWaitReady(Id int) {
+	// Wait for transactions to finish
+	fmt.Printf("Waiting for transactions to complete before destroying.")
+	s := services.GetVirtualGuestService(gs.sess).Id(Id)
+
+	// Delay to allow transactions to be registered
+	time.Sleep(5 * time.Second)
+
+	for transactions, _ := s.GetActiveTransactions(); len(transactions) > 0; {
+		fmt.Print(".")
+		time.Sleep(5 * time.Second)
```

Comment:
I guess we need TODO to make it configurable or use the notification mechanism to optimize the process instead of hardcoded waiting...